packages/suite/src/actions/suite/metadataActions.ts

```diff
@@ -730,6 +1022,36 @@ export const init =
 
             const providerResult = await dispatch(initProvider());
             if (!providerResult) {
+                dispatch(abortInit());
+                return false;
+            }
+        }
+
+        // 4. get the metadata minimum version to determine a migration is needed
+        const metadataFilesList = await dispatch(getMetadataFilesList());
+        const metadataFilesMimimumVersion = getMetadataFilesMinimumVersion(metadataFilesList);
+
+        if (getState().metadata.initiating && !metadataFilesMimimumVersion) {
+            dispatch(abortInit());
+            return false;
+        }
+
+        // if yes, add metadata keys to accounts
+        if (getState().metadata.initiating) {
+            dispatch(syncMetadataKeys());
+        }
+
+        // if there were some old files - migrate if needed
+        const isMinimumFileVersionLatest =
+            metadataFilesMimimumVersion === METADATA_ENCODING_VERSION;
+        if (getState().metadata.initiating && !isMinimumFileVersionLatest) {
+            try {
+                const missingFileNames = dispatch(getMissingFiles(metadataFilesList));
```

Comment:
total nit:  `missingFileNames` is of type `MissingFileInfo[]`. Maybe some better variable name, I don't know `missingFilesInfo`.  Feel totally free to dismiss

Reply:
done